jina/drivers/evaluate.py

```diff
@@ -0,0 +1,58 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Iterable
+from .helper import DocGroundtruthPair
+from . import BaseExecutableDriver
+
+
+class BaseEvaluationDriver(BaseExecutableDriver):
+    def __init__(self, executor: str = None,
+                 method: str = 'evaluate',
+                 *args,
+                 **kwargs):
+        super().__init__(executor, method, *args, **kwargs)
+
+    def __call__(self, *args, **kwargs):
+        assert len(self.req.docs) == len(self.req.groundtruths)
+        docs_groundtruths = [DocGroundtruthPair(doc, groundtruth) for doc, groundtruth in
+                             zip(self.req.docs, self.req.groundtruths)]
+        self._traverse_apply(docs_groundtruths, *args, **kwargs)
+
+    def _apply_all(self, docs_groundtruths: Iterable[DocGroundtruthPair],
+                   context_doc_groundtruth: DocGroundtruthPair,
+                   *args,
+                   **kwargs) -> None:
+        pass
+
+
+class RankingEvaluationDriver(BaseEvaluationDriver):
+    """Drivers inherited from this Driver will bind :meth:`evaluate` by default
+    """
+
+    def __init__(self,
+                 id_tag='id',
+                 *args,
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+        self.id_tag = id_tag
+
+    @property
+    def id(self):
+        if self.pea:
+            return self.pea.name
+        else:
+            return self.__class__.__name__
+
+    def _apply_all(self,
+                   docs_groundtruths: Iterable[DocGroundtruthPair],
+                   *args,
+                   **kwargs) -> None:
+        for doc_groundtruth in docs_groundtruths:
+            doc = doc_groundtruth.doc
+            groundtruth = doc_groundtruth.groundtruth
+            evaluation = doc.evaluations.add()
+            matches_ids = list(map(lambda x: x.tags[self.id_tag], doc.matches))
```

Comment:
This should rather be a list comprehension, then a `lambda` function. In my opinion 
```
matches_ids = [x.tags[self.id_tag] for x in doc.matches] 
```
is way better readable.

Reply:
Map with lambda is slower than comprehension as per a SO discussion